spring-cloud-gcp-starters/spring-cloud-gcp-starter-sql/src/main/java/org/springframework/cloud/gcp/sql/autoconfig/GcpCloudSqlAutoConfiguration.java

```diff
@@ -131,34 +132,34 @@ public CloudSqlJdbcInfoProvider cloudSqlJdbcInfoProvider(SQLAdmin sqlAdmin,
 	}
 
 	@Bean
+	@ConfigurationProperties(prefix = "spring.datasource")
```

Comment:
A problem with this approach is that we still don't have access to those low level knobs, like max pool size, etc.

I was thinking we could leverage [DataSourceConfiguration](https://github.com/spring-projects/spring-boot/blob/master/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/jdbc/DataSourceConfiguration.java) somehow, but it isn't looking like we can right off the bat. e.g., the class is package protected, not integrated with our `CloudSqlJdbcInfoProvider`, etc. But maybe there's still a way to use [DataSourceProperties](https://github.com/spring-projects/spring-boot/blob/master/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/jdbc/DataSourceProperties.java) with our stuff, to support a more customized Hikari `DataSource` (and possibly others).

Reply:
@artembilan might have an opinion about this